### PR TITLE
Add tech stack for java

### PR DESCRIPTION
**Pull request by Barry-RG**
- [x] Are you happy for the content of this change to be publicly visible?
- [x] Are all new or updated entries marked as `isNew` = `true`?
- [x] Does `radar.csv` render correctly when viewed on Github?
- [x] Does the updated [tech radar](https://radar.thoughtworks.com/?sheetId=https%3A%2F%2Fraw.githubusercontent.com%2Fred-gate%2Ftech-radar%2Fflyway%2Fradar.csv) display as you expect?

### CODE REVIEW

**Comment by fffej** [comment]
@Barry-RG Have you got the right line endings settings? It's difficult to use the diff at the moment and I _think_ there's a typo in one of the categories so it doesn't display correctly.

**Comment by Barry-RG** [comment]
@fffej sorry, didn't realize what was causing that. Excel stripped the quotes out. Thought it was just github being strange

File: radar.csv [hunk 1]
```diff
@@ -1,8 +1,8 @@
-﻿name,ring,quadrant,isNew,description
+name,ring,quadrant,isNew,description
 "C#",adopt,"Languages and Frameworks",true,"<a href='https://docs.microsoft.com/en-us/dotnet/csharp/'>C#</a> is our language of choice. It's strongly-typed, blends functional and object-oriented programming and works on all the OS we need to support."
 "TypeScript",adopt,"Languages and Frameworks",true,"<a href='https://www.typescriptlang.org/'>TypeScript</a> is our default choice for running code in the browser. It's strongly-typed and smooths out the rough bits of JavaScript."
 "Go",trial,"Languages and Frameworks",true,"<a href='https://golang.org/'>Go</a> is the natural language to use when extending Kubernetes with operators."
-"Java",trial,"Languages and Frameworks",true,"Redgate purchased Flyway, written in <a href='https://www.oracle.com/java/'>Java</a>. Java is a plausible choice for new products (due to its rich ecosystem), but we default to C#."
+"Java 8",adopt,"Languages and Frameworks",true,"Redgate purchased Flyway, written in <a href='https://www.oracle.com/java/'>Java</a>. Java is a plausible choice for new products (due to its rich ecosystem), but we default to C#."
```
Comment (dnlegge) [inline radar.csv:5]: Is Java 8 not now deprecated/eol?

**Comment by fffej** [comment]
@Barry-RG We made some changes to the tech radar a year or so back when we removed libraries from the radar; instead trying to focus on standardizing the "glue".

I guess JDBC is an important one to call out (we should be consistent across products on that), but JANSI, GSON and JUnit feel like something a team could flex on without affecting anyone else? If you agree, could you remove them?

File: radar.csv (hunk 1 above)
Comment (Barry-RG) [reply radar.csv:5]: Sadly for Java 8 that's not the case. Java 8 has extended support until 2030. Unfortunately, due to changing in Oracle business strategies at the time they set their paid for extended support for Java 8 till 2030, which a lot of companies adopted, which outlasts even the latest LTS version of Java (17) which will only get extended support till 2029. This makes moving forward from Java 8 very difficult in the commercial space.

Comment (Barry-RG) [reply radar.csv:5]: In regards to your march 2022 date,, that's free updates. And in regards to the security only be 3 more years, I do not know where that has come from. 
Please see https://www.oracle.com/java/technologies/java-se-support-roadmap.html

**Comment by Barry-RG** [comment]
@fffej JANSI and GSON that may be fair. However, I would stick with having JUnit there as its precedent there for it to be included as part of a Java stack as its a usually good to have a test framework specified.

File: radar.csv (hunk 1 above)
Comment (dnlegge) [reply radar.csv:5]: Fair enough - thanks for clarifying.

**Comment by fffej** [comment]
@Barry-RG Sounds good - if you could merge the conflicts and drop JANSI/GSON then I'll press merge 😄

**Review by fffej — APPROVED** [review]

**Pull request merged**